hydradx/tests/test_omnipool_amm.py

```diff
@@ -1679,4 +1679,298 @@ def test_oracle_one_block_with_swaps(liquidity: list[float], lrna: list[float],
+                minimum=lrna_fee_params['minimum'],
+                amplification=lrna_fee_params['amplification'],
+                raise_oracle_name=lrna_fee_params['raise_oracle_name'],
+                decay=lrna_fee_params['decay'],
+                fee_max=lrna_fee_params['fee_max'],
+            ),
+            'USD': dynamicadd_lrna_fee(
+                minimum=lrna_fee_params['minimum'],
+                amplification=lrna_fee_params['amplification'],
+                raise_oracle_name=lrna_fee_params['raise_oracle_name'],
+                decay=lrna_fee_params['decay'],
+                fee_max=lrna_fee_params['fee_max'],
+            ),
+            'DOT': dynamicadd_lrna_fee(
+                minimum=lrna_fee_params['minimum'],
+                amplification=lrna_fee_params['amplification'],
+                raise_oracle_name=lrna_fee_params['raise_oracle_name'],
+                decay=lrna_fee_params['decay'],
+                fee_max=lrna_fee_params['fee_max'],
+            ),
+        },
+        last_oracle_values={
+            'oracle': copy.deepcopy(init_oracle)
+        },
+        last_lrna_fee=copy.deepcopy(init_lrna_fees),
+        last_asset_fee=copy.deepcopy(init_asset_fees),
+    )
+
+    trader_holdings = {'HDX': 1000000000, 'USD': 1000000000, 'LRNA': 1000000000, 'DOT': 1000000000}
+
+    initial_state = GlobalState(
+        pools={'omnipool': initial_omnipool},
+        agents={
+            'trader': Agent(
+                holdings=trader_holdings,
+                trade_strategy=constant_swaps(
+                    pool_id='omnipool',
+                    sell_quantity=trade_size,
+                    sell_asset='USD',
+                    buy_asset='DOT'
+                )
+            ),
+        }
+    )
+
+    events = run.run(initial_state=initial_state, time_steps=2, silent=True)
+
+    # test non-empty block fee dynamics
+
+    omnipool = events[1].pools['omnipool']
+    prev_lrna_fees = events[0].pools['omnipool'].last_lrna_fee
+    prev_asset_fees = events[0].pools['omnipool'].last_fee
+    omnipool_oracle = omnipool.oracles['oracle']
+    for tkn in ['HDX', 'USD', 'DOT']:
+        x = (omnipool_oracle.volume_out[tkn] - omnipool_oracle.volume_in[tkn]) / omnipool_oracle.liquidity[tkn]
+
+        df = -lrna_fee_params['amplification'] * x - lrna_fee_params['decay']
+        expected_lrna_fee = min(max(prev_lrna_fees[tkn] + df, lrna_fee_params['minimum']), lrna_fee_params['fee_max'])
+        if omnipool.last_lrna_fee[tkn] != pytest.approx(expected_lrna_fee, rel=1e-15):
+            raise AssertionError('LRNA fee is not correct.')
+
+        df = asset_fee_params['amplification'] * x - asset_fee_params['decay']
+        expected_asset_fee = min(max(prev_asset_fees[tkn] + df, asset_fee_params['minimum']), asset_fee_params['fee_max'])
+        if omnipool.last_fee[tkn] != pytest.approx(expected_asset_fee, rel=1e-15):
+            raise AssertionError('Asset fee is not correct.')
```

Comment:
newline at end. It's one of those things Pycharm will insist on

Reply:
It has one, github isn't showing it